python/tvm/relay/frontend/tflite.py

```diff
@@ -289,6 +290,25 @@ def convert_concatenation(self, op):
             out = self.convert_fused_activation_function(out, fused_activation_fn)
         return out
 
+    def convert_add(self, op):
+        """Convert TFLite add"""
+        try:
+            from tflite.Operator import Operator
+        except ImportError:
+            raise ImportError("The tflite package must be installed")
+
+        assert isinstance(op, Operator)
+        input_tensors = self.get_input_tensors(op)
+        assert len(input_tensors) == 2, "input tensors length should be 2"
+
+        lhs_tensor = input_tensors[0]
+        rhs_tensor = input_tensors[1]
+        lhs_expr = self.get_expr(lhs_tensor.tensor_idx)
+        rhs_expr = self.get_expr(rhs_tensor.tensor_idx)
```

Comment:
If `rhs_expr` is constant value, this will have problem. We should consider this case. (TFLite supports it).

Reply:
Thank you for review. Could you please let me know more details or reference-able code?

Reply:
For example, `lhs_expr` is from one op's output, it could use input_tensor. However, if `rhs_expr` is constant values, like conv2d's weight, it couldn't use input_tensor, should be `get_tensor_value` like we get conv2d's weight.

Reply:
Thanks for your kind explanation. Can you advise me if I understand correctly?
```suggestion
lhs_tensor = input_tensors[0]
rhs_tensor = input_tensors[1]
lhs_expr = self.get_expr(lhs_tensor.tensor_idx)
if isinstance(rhs_tensor, TensorWrapper):
    rhs_expr = self.get_expr(rhs_tensor.tensor_idx)
else:
    rhs_expr = self.exp_tab.new_const(self.get_tensor_value(rhs_tensor),
                                      self.get_tensor_type_str(rhs_tensor.tensor.Type()))
```

Reply:
Yes. However,  let us imagine the constant value's shape is not the same as `lhs_expr`. For example: `lhs_expr` is [1, 3, 224, 224]. (we use `NCHW` layout). `rhs_expr` is [3]. (Because original TFLite layout is `NHWC` layout. So broadcast_add is no problem, it will be [1, 224 224, 3] add [1, 1, 1, 3]. But we will be [1, 3, 224, 224] add [1, 1, 1, 3], it is wrong) So, we should handle the axis very carefully.  if both are Tensor, it is no problem (we have converted into NCHW). But if it is constant values, we haven't translated. Alright, If our translator is `NHWC` layout. it is no problem just to add. But we are `NCHW` layout currently, we should be careful about it. You could refere: https://github.com/dmlc/tvm/issues/2519 to see why we are using `NCHW` layout in TFLite currently. Maybe we will use `NHWC` layout in the future. Sorry for introducing inconvience currently when we handle these ops. You could refer `resize` / `squeeze` to find how to handle.

Reply:
Hi @FrozenGene, I understood your concern regarding layout problem. I hope to explain why I only assumed the tensor input. In case of the Tensorflow-Lite frontend, because the TOCO converter will fuse the ADD operator which has constant value, I think we can assume the remained ADD operators always will receive two tensors. I've written code based on this understanding. I hope to know your opinion regarding this. :)

Reply:
However, in fact, I met one model TFlite doesn't fuse actually. I think you could do one experiment, construct one TFLite model just one op `add` and make `rhs` be constant values. It shouldn't be fused, otherwise our model shouldn't met `add` constant values.

Reply:
The corner case is good point. That is make sense. I made dummy.tflite and reproduced your experience. The interesting point is `if isinstance(rhs_tensor, TensorWrapper):` couldn't distinguish constant input. I'll investigate more - if you have any other idea to distinguish, please let me know, that will be helpful.

Reply:
You could do like this: 

- Add one method `has_expr` in `ExprTable`, the argument is the name of expr. 

- We could use `has_expr` to judge whether it is Tensor or not. The argument to `has_expr` could use `get_tensor_name` to get.

Then you could distinguish.

Reply:
Hi @FrozenGene, Thanks! I've updated and tested the code with reference to `squeeze`.

Reply:
Thanks @gomida. Could you add unit testing for `Add` like other ops? Not just whole model testing.

Reply:
I've added unit test for ADD operator. To test the 3-dim layout, the function of output_need_transpose_nchw flag has expanded.

Current code has a little lack of coverage. I could not make the test case for constant input. Changing `array_ops.placeholder` to` constant_op.constant` seems differs from creating a dummy.tflite. If you have idea, please advise me.

Reply:
I think you could use
```python
from tensorflow.python.framework import ops
constant_values = ops.convert_to_tensor(SOME_NUMPY_VALUES)
```

Reply:
Your suggestion working well :)